sdk/keyvault/Azure.Security.KeyVault.Shared/src/ChallengeBasedAuthenticationPolicy.cs

```diff
@@ -13,24 +13,24 @@ namespace Azure.Security.KeyVault
     internal class ChallengeBasedAuthenticationPolicy : BearerTokenChallengeAuthenticationPolicy
     {
         private static ConcurrentDictionary<string, AuthorityScope> _scopeCache = new ConcurrentDictionary<string, AuthorityScope>();
+        private ConcurrentDictionary<HttpMessage, RequestContent> _contentCache = new ConcurrentDictionary<HttpMessage, RequestContent>();
```

Comment:
This feels wrong. We shouldn't the concurrent dict just to pass data down the call stack.

Reply:
why don't you cache this on the message itself instead?

Reply:
Caching on the message felt more fragile to me, since any pipeline in the chain could modify the message. But I could be convinced otherwise :)

Reply:
Discussed offline - we are now caching on the message.